### PR TITLE
Add helper to build virtualenv and vscode configuration

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

- Add tox configuration that allows to build a venv for development and to
  format source files. To make use of it add appropriate Makefile targets.
- Drop unneeded requirements from requirements.txt and
  test-requirements.txt and add needed requirements.
- Add tox-ansible configuration that allows to run tests with all
  required ansible and python versions. Add Makefile targets to make use
  of it.
- Add vscode configuration that allows to develop, test and debug the
  collection.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: hack/build-venv.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Create and activate venv
+python -m venv .venv
+source .venv/bin/activate
```
Comment (lyarwood) [inline hack/build-venv.sh:5]: Might be easier to use [`tox`](https://tox.wiki/en/4.14.2/) for all of this?

For example - https://opendev.org/openstack/ansible-collections-openstack/src/branch/master/tox.ini

File: tests/unit/requirements.txt [hunk 2]
```diff
@@ -1,5 +0,0 @@
-jinja2
-jsonpatch
-kubernetes>=28.1.0
-PyYAML>=3.11
```
Comment (lyarwood) [inline tests/unit/requirements.txt:4]: I thought `PyYAML` and `kubernetes` were still requirements?

Comment (lyarwood) [reply tests/unit/requirements.txt:4]: Ignore me this is in `tests/unit`.

File: hack/build-venv.sh (hunk 1 above)
Comment (0xFelix) [reply hack/build-venv.sh]: Updated to use tox :+1:

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,22 @@
+[tox]
```
Comment (lyarwood) [inline tox.ini:1]: Is `[tox-ansible.ini](https://github.com/kubevirt/kubevirt.core/pull/78/files#diff-56bafa1debb1532b01b6817dcc3bcaa50bd82aa8f85b6e585117c89add5788be)` supposed to be empty?

Comment (0xFelix) [reply tox.ini:1]: Yes, to use default configuration an empty file is enough. See [here](https://github.com/ansible/tox-ansible?tab=readme-ov-file#usage).

**Comment by lyarwood** [comment]
@jcanocan can you also take a look at this please?

**Comment by jcanocan** [comment]
This is working perfectly fine. Great job @0xFelix! Thanks a lot!
/lgtm

**Pull request merged**